Tasks/GoToolV0/gotool.ts

```diff
@@ -119,20 +131,117 @@ function setGoEnvironmentVariables(goRoot: string) {
+        return false;
+    } catch {
+        // Invalid URL string => treat as custom to avoid unintended metadata fetch
+        return false;
+    }
+}
+
+function includesLatestKeyword(url?: string): boolean {
+    if (!url) return false;
+    return url.toLowerCase().includes('latest');
+}
+
+async function getLatestPatchFromGoDev(majorMinorVersion: string): Promise<string> {
+    const normalized = majorMinorVersion.replace(/^go/i, '').replace(/^v/i, '');
+    const parts = normalized.split('.');
+    if (parts.length < 2) {
+        throw new Error(`Invalid major.minor version: ${majorMinorVersion}`);
+    }
+    const major = parseInt(parts[0], 10);
+    const minor = parseInt(parts[1], 10);
+
+    const metadataUrl = 'https://go.dev/dl/?mode=json&include=all';
+    let jsonPath: string;
+    try {
+        tl.debug(`Downloading Go releases metadata from ${metadataUrl}`);
+        jsonPath = await toolLib.downloadTool(metadataUrl);
+    } catch (error) {
+        throw new Error(`Failed to download Go releases metadata: ${error.message}`);
+    }
+    const raw = fs.readFileSync(jsonPath, 'utf8');
+
+    let releases: any[];
+    try {
+        releases = JSON.parse(raw);
+    } catch (e) {
+        throw new Error('Failed to parse Go releases metadata from go.dev');
+    }
+
+    let maxPatch = -1;
+    for (const rel of releases) {
+        if (!rel || !rel.version || rel.stable !== true) continue;
+        const v: string = String(rel.version);
+        const m = /^go(\d+)\.(\d+)(?:\.(\d+))?$/i.exec(v);
+        if (!m) continue;
+        const maj = parseInt(m[1], 10);
+        const min = parseInt(m[2], 10);
+        const pat = m[3] ? parseInt(m[3], 10) : 0;
+        if (maj === major && min === minor) {
+            if (pat > maxPatch) maxPatch = pat;
+        }
+    }
+
+    if (maxPatch < 0) {
+        throw new Error(`Could not find a stable patch version for ${major}.${minor} from go.dev`);
+    }
+    return `${major}.${minor}.${maxPatch}`;
+}
+
+async function resolveVersionAndCache(version: string, baseUrl?: string): Promise<{ filenameVersion: string, cacheVersion?: string }> {
+    const v = version.trim().replace(/^v/i, ''); // Remove leading 'v' if present to account for user mistakes
+    const official = isOfficialBaseUrl(baseUrl);
+
+    if (official) {
+        if (isSemverWithPatch(v)) {
+            return { filenameVersion: v, cacheVersion: v };
+        }
+        if (hasMajorMinorOnly(v)) {
+            const resolved = await getLatestPatchFromGoDev(v);
+            return { filenameVersion: resolved, cacheVersion: resolved };
+        }
+        // Fallback to semantic conversion for other formats (e.g., "1")
+        throw new Error("For official Go URLs, the version spec must be in 'major.minor' (e.g., '1.10') or 'major.minor.patch' (e.g., '1.10.2') format.");
+    } else {
+        const urlHasLatest = includesLatestKeyword(baseUrl);
+        if (isSemverWithPatch(v)) {
```

Comment:
Minor+Patch isn't specific for the Microsoft build of Go, there's also the Revision. Generally `1.x.y-z`. https://github.com/microsoft/go/blob/microsoft/main/eng/doc/Downloads.md#downloading-a-specific-version

(At least, this code seems to imply that `1.x.y` is expected to be immutable.)